remotespark/utils/configuration.py

```diff
@@ -89,14 +91,38 @@ def session_configs():
     return {}
 
 
-@_override
+def _credentials_override(f):
+    """A decorator that provide special handling for credentials. It still calls _override().
+    If 'base64_password' in config is set, it will base64 decode it and returned in return value's 'password' field.
+    If 'base64_password' is not set, it will fallback to to 'password' in config.
+    """
+    def ret():
+        credentials = _override(f)()
+        base64_decoded_credentials = {k: credentials.get(k) for k in ('username', 'password', 'url')}
+        base64_password = credentials.get('base64_password')
+        if base64_password is not None:
+            try:
+                base64_decoded_credentials['password'] = base64.b64decode(base64_password).decode()
+            except Exception:
+                exception_type, exception, traceback = sys.exc_info()
+                msg = "base64_password for %s contains invalid base64 string: %s %s" % (f.__name__, exception_type, exception)
+                raise RuntimeError(msg)
```

Comment:
Please look at how we do error handling. Only Exceptions of a certain type are handled and bubbled up to users. Look at `handle_expected_exceptions` inside `utils\exceptions.py`

I would say misconfiguration is an expected exception (not a bug) and you should create an exception for this type of error.